### PR TITLE
Hotfix for ImplicitActiveSheetReference inspection

### PR DESCRIPTION
**Pull request by retailcoder**
This patch makes the ImplicitActiveSheetReference inspection tests to now correctly set the document module supertypes in the test code, and adds a test that confirms the fix for the referenced issue.

### CODE REVIEW

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ImplicitActiveSheetReferenceInspection.cs [hunk 1]
```diff
@@ -47,10 +47,13 @@ public ImplicitActiveSheetReferenceInspection(IDeclarationFinderProvider declara
             : base(declarationFinderProvider)
         {}
 
+        protected override string[] GlobalObjectClassNames => new[] { "Global", "_Global", };
+
         protected override bool IsResultReference(IdentifierReference reference, DeclarationFinder finder)
         {
-            return !(Declaration.GetModuleParent(reference.ParentNonScoping) is DocumentModuleDeclaration document)
+            var result = !(Declaration.GetModuleParent(reference.ParentNonScoping) is DocumentModuleDeclaration document)
```
Comment (MDoerner) [inline Rubberduck.CodeAnalysis/Inspections/Concrete/ImplicitActiveSheetReferenceInspection.cs:54]: Why the intermediate variable?

File: RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs [hunk 2]
```diff
@@ -23,8 +30,13 @@ public void ImplicitActiveSheetReference_ReportsRange()
...
 ";
-            var modules = new(string, string, ComponentType)[] { ("Class1", inputCode, ComponentType.ClassModule) };
-            Assert.AreEqual(1, InspectionResultsForModules(modules, ReferenceLibrary.Excel).Count());
+            var modules = new(string, string, ComponentType)[] 
+            {
+                ("ThisWorkbook", string.Empty, ComponentType.Document),
+                ("Sheet1", string.Empty, ComponentType.Document),
+                ("Class1", inputCode, ComponentType.ClassModule)
+            };
+            Assert.AreEqual(1, InspectionResultsForModules(modules, ReferenceLibrary.Excel, DefaultDocumentModuleSupertypeNames).Count());
```
Comment (MDoerner) [inline RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs:39]: Why this change? Neither `Sheet1` nor `ThisWorkbook` is relevant to the test.

File: RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs [hunk 3]
```diff
@@ -106,21 +125,33 @@ public void ImplicitActiveSheetReference_ReportsInWorkbookModules()
...
+            const string inputCode =
+                @"Sub foo()
+    Dim sh As Worksheet
+    Set sh = Sheet1
+    Debug.Print sh.Cells(1, 1)
+End Sub
+";
+            var modules = new (string, string, ComponentType)[]
+            {
+                ("ThisWorkbook", string.Empty, ComponentType.Document),
```
Comment (MDoerner) [inline RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs:150]: Here, `ThisWorkbook` is not relevant to the test.

File: RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs [hunk 4]
```diff
@@ -106,21 +125,33 @@ public void ImplicitActiveSheetReference_ReportsInWorkbookModules()
...
+            };
+            Assert.AreEqual(1, InspectionResultsForModules(modules, ReferenceLibrary.Excel, DefaultDocumentModuleSupertypeNames).Count());
+        }
 
-                Assert.AreEqual(1, inspectionResults.Count());
-            }
+        [Test]
+        [Category("Inspections")]
+        public void ImplicitActiveSheetReference_NoResultForWorksheetVariable()
+        {
```
Comment (MDoerner) [inline RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs:140]: Can we please have a corresponding test for `ImplicitContainingSheetReferenceInspection` as well?

**Review by MDoerner — COMMENTED** [review]
Why are there all these additions of irrelevant sheets and workbooks to the tests? I would expect tests to only bring things into scope what is relevant to the test.

Moreover, could you please also test the variable issue for the containing sheet inspection?

Finally, could yousquash a little xml-doc correction int this PR? The doc should say outside worksheet modules for the active sheet inspection. Currently, it says inside, because if a copy-paste error on my side.

File: RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs (hunk 3 above)
Comment (retailcoder) [reply RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs:150]: It's not, but any real-life VBA project hosted in Excel is always going to have a `ThisWorkbook` and at least one `Worksheet` module. Not including them would make the tests work in a hypothetical scenario that cannot happen (e.g. an Excel-hosted VBA project without a `ThisWorkbook` module).

File: RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs (hunk 4 above)
Comment (retailcoder) [reply RubberduckTests/Inspections/ImplicitActiveSheetReferenceInspectionTests.cs:140]: Good call, will add.

File: Rubberduck.CodeAnalysis/Inspections/Concrete/ImplicitActiveSheetReferenceInspection.cs (hunk 1 above)
Comment (retailcoder) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/ImplicitActiveSheetReferenceInspection.cs:54]: Debugger artifact, needed to break here and inspect `document` supertypes; will remove.
